### PR TITLE
dnsdist: add support for user defined metrics

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description

This PR adds a few functions to declare and manipulate user-defined metrics. Currently supports `gauge` and `counter` types with set, increment and decrement operations.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/dnsdistdist/docs/reference/custommetrics.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  Get metric value
+
+  :param str name: The name of the metric
+
+.. function:: setMetric(name, value) -> double
+
+  .. versionadded:: 1.x
+
+  Decrement counter by one, will issue an error if the metric is not declared or not a ``counter``
```
Comment (rgacogne) [inline pdns/dnsdistdist/docs/reference/custommetrics.rst:52]:
```suggestion
  Set the new value, will issue an error if the metric is not declared or not a ``gauge``
```

**Review by rgacogne — CHANGES_REQUESTED** [review]
Looks good, thanks! A couple things need to be tied up but the general logic looks great.

**Comment by Habbie** [comment]
> I cannot reproduce the CI failure locally, though :-/

Me neither :(

**Review by rgacogne — APPROVED** [review]
Looks good to me, I'll do more tests once the CI is green :+1:

**Comment by chbruyand** [comment]
Thanks for the comprehensive review!

**Pull request merged**